### ISSUE TITLE
Make use of DifferentialEquations.jl

### ISSUE THREAD

**Issue by ufechner7** [ModiaSim/ModiaMath.jl#1]
Using the abstract solver interface as defined in DifferentialEquations.jl would make it possible to use different solvers in the future. In particular the option to use solvers written in Julia that support all Julia data types would create new possibilities.

**Comment by MartinOtter** [comment]
Yes, this is already planned (mentioned at beginning of README.md file and in the "Plans for version 1.0").

**Comment by ufechner7** [comment]
Nice, but I think having an issue for it is a good place to discuss the approach and the progress. To implement this, the main place to make changes is simulate.jl. Is that correct? If the needed changes are implemented here, would it also be needed to modify Modia.jl?
